docs/design/hybrid_kv_cache_manager.md

```diff
@@ -0,0 +1,230 @@
+# Hybrid KV Cache Manager
+
+Note: This doc is written based on commit 458e74eb907f96069e6d8a4f3c9f457001fef2ea on 2025-08-11. This feature is still in its early stage and things may change.
```

Comment:
Could you change the hash to a permalink?

Also, the edit date is now included at the bottom of every page so it's not necessary to mention it explicitly.